lookup.go

```diff
@@ -0,0 +1,42 @@
+package twilio
+
+import (
+	"context"
+	"net/url"
+)
+
+const phoneNumbersPath = "PhoneNumbers"
+
+type LookupPhoneNumbersService struct {
+	client *Client
+}
+
+type CallerLookup struct {
+	CallerName string `json:"caller_name"`
+	CallerType string `json:"caller_type"`
+	ErrorCode  string `json:"error_code"`
```

Comment:
Are you sure these (and the ErrorCode parameters below) are returned as strings? Is there a way we can test this? If you can't trigger an API error (a short phone number maybe?), you may want to check a typed library like the Java library (if it supports this API) .

Reply:
CallerName and CallerType for sure, check the example response at https://www.twilio.com/lookup

For ErrorCode, the docs aren't clear, but if i look at error responses where they return a `code`, that code it's an integer value. Also, the error reference only talks about integer codes, so i think we can make it an int. I'm on it.

Reply:
That said, the Java SDK defines the `callerName` and `carrier` types as Map<String,String> so maybe we can leave the error code in there as string: https://github.com/twilio/twilio-java/blob/6c18bdae18831808014e82504b151fbfdf81b5eb/src/main/java/com/twilio/rest/lookups/v1/PhoneNumber.java#L119

What do you think?

Reply:
And BTW, about testing it, checking a wrong number returns an error response instead of this one, so that may be different: 

Using https://www.twilio.com/lookup: 

```
curl -XGET "https://lookups.twilio.com/v1/PhoneNumbers/415701231?Type=carrier&Type=caller-name" -u "{AccountSid}:{AuthToken}"
```
(shorter/invalid number)

Returns: 

```
{
    "status": 404,
    "more_info": "https://www.twilio.com/docs/errors/20404",
    "code": 20404,
    "message": "The requested resource /PhoneNumbers/415701231 was not found"
}
```

Reply:
It would be best to try to get an example where this gets returned from the API if possible.

Can you send a request to help@twilio.com and cc me on it? Just say we're having trouble figuring out what the types should be here and it would be helpful if we could see an example of how the API returns them.

Reply:
maybe Java has smarter type casting, or maybe it breaks when you try to decode a response- both seem possible

Reply:
On it.

Reply:
404 seems like the wrong code there... would probably rather return a 400, but oh well.